packages/functional/tests/ui/desktop/e2e/coupon.spec.ts

```diff
@@ -0,0 +1,86 @@
+
+const sampleProduct = '[Sample] Able Brewing System';
+const totalAmount = '[data-test="cart-total"] [data-test="cart-price-value"]';
+const couponCode = 'OFF25';
+
+test.beforeEach(async ({ page }) => {
+  await page.goto('/');
+  await page.getByLabel('Main').getByRole('link', { name: 'Kitchen' }).click();
+
+  await expect(page.getByRole('heading', { level: 1, name: 'Kitchen' })).toBeVisible();
+});
+
+test('Add coupon code', async ({ page }) => {
+  await ProductActions.addProductToCart(page, sampleProduct);
+
+  await page.getByRole('link', { name: 'Cart Items 1' }).click();
+
+  await expect(page.getByRole('heading', { level: 1, name: 'Your cart' })).toBeVisible();
+  await expect(page.getByRole('link', { name: 'Proceed to checkout' })).toBeVisible();
+
+  await page.getByRole('button', { name: 'Add' }).nth(1).click();
+
+  const couponCodeBox = page.getByPlaceholder('Enter your coupon code');
+
+  await couponCodeBox.fill(couponCode);
+  await couponCodeBox.press('Enter');
+
+  await expect(page.getByText(`Coupon (${couponCode})`)).toBeVisible();
+});
+
+test('Coupon code is required', async ({ page }) => {
+  await ProductActions.addProductToCart(page, sampleProduct);
+
+  await page.getByRole('link', { name: 'Cart Items 1' }).click();
+
+  await expect(page.getByRole('heading', { level: 1, name: 'Your cart' })).toBeVisible();
+  await expect(page.getByRole('link', { name: 'Proceed to checkout' })).toBeVisible();
+
+  await page.getByRole('button', { name: 'Add' }).nth(1).click();
+
+  const couponCodeBox = page.getByPlaceholder('Enter your coupon code');
+
+  await couponCodeBox.fill('');
+  await couponCodeBox.press('Enter');
+
+  await expect(page.getByText('Please enter a coupon code.')).toBeVisible();
+});
+
+test('Coupon code fails', async ({ page }) => {
+  await ProductActions.addProductToCart(page, sampleProduct);
+
+  await page.getByRole('link', { name: 'Cart Items 1' }).click();
+
+  await expect(page.getByRole('heading', { level: 1, name: 'Your cart' })).toBeVisible();
+  await expect(page.getByRole('link', { name: 'Proceed to checkout' })).toBeVisible();
+
+  await page.getByRole('button', { name: 'Add' }).nth(1).click();
+
+  const couponCodeBox = page.getByPlaceholder('Enter your coupon code');
+
+  await couponCodeBox.fill('INCORRECT_CODE');
+  await couponCodeBox.press('Enter');
+
+  await expect(page.getByText('The coupon code you entered is not valid.')).toBeVisible();
+});
+
+test('Apply coupon on checkout', async ({ page }) => {
+  await ProductActions.addProductToCart(page, sampleProduct);
+
+  await page.getByRole('link', { name: 'Cart Items 1' }).click();
+
+  await expect(page.getByRole('heading', { level: 1, name: 'Your cart' })).toBeVisible();
+
+  await page.getByRole('link', { name: 'Proceed to checkout' }).click();
+
+  await expect(page.getByRole('link', { name: 'Coupon/Gift Certificate' })).toBeVisible();
+  await expect(page.locator(totalAmount)).toContainText('$225.00');
```

Comment:
Couldn't find a better way to assert text in a span.

Reply:
Does `page.getByText('$225.00')` not work?

Reply:
It's at 3 different locations as it resolved to 3 elements and we might also want to know where we are looking at just so that it's at the right place so, I went with the locator.

Reply:
What about `page.getByText('Grand total $225.00')`?